### PR TITLE
feat: Add 2 main functions table_schema and table_names for Parquet Chunk to pay a foundation for querying it

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #

First part of #1082. This PR focus on adding 2 functions table_schema and table_names for Parquet Chunk

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: parquet_file/src/table.rs [hunk 1]
```diff
@@ -47,4 +73,24 @@ impl Table {
...
+                self.table_schema.project(&columns)
+            }
+        })
+    }
+
+    pub fn matches_predicate(&self, timestamp_range: &Option<TimestampRange>) -> bool {
+        match (self.timestamp_range, timestamp_range) {
+            (Some(a), Some(b)) => !a.disjoint(b),
+            (None, Some(_)) => false, /* If this chunk doesn't have a time column it can't match */
+            // the predicate
```
Comment (mkmik) [inline parquet_file/src/table.rs:92]: Has this comment been chopped up by your editor?

**Review by mkmik — APPROVED** [review]
Looks good to me (modulo minor comment about a comment formatting)

File: parquet_file/src/table.rs (hunk 1 above)
Comment (NGA-TRAN) [reply parquet_file/src/table.rs:92]: Yes. Cargo fit --all did it

File: server/src/db/chunk.rs [hunk 2]
```diff
@@ -165,9 +171,7 @@ impl PartitionChunk for DBChunk {
 
                 chunk.table_names(&rb_predicate, &BTreeSet::new())
             }
-            Self::ParquetFile { .. } => {
-                unimplemented!("parquet file not implemented for scan_data")
-            }
+            Self::ParquetFile { chunk, .. } => chunk.table_names(predicate.range).collect(),
```
Comment (alamb) [inline server/src/db/chunk.rs:174]: 7drHiqr

So exciting

**Pull request merged**